dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/util/FileUtil.java

```diff
@@ -749,16 +746,14 @@ public static String getBatchFilesDownloadUrlPath(List<Long> fileIds, boolean gu
         return fileDownloadUrl;
     }
 
-    public static String getDownloadWholeDatasetUrlPath(DatasetVersion dsv, boolean guestbookRecordsAlreadyWritten, ApiBatchDownloadType downloadType) {
+    public static String getDownloadWholeDatasetUrlPath(DatasetVersion dsv, ApiBatchDownloadType downloadType) {
```

Comment:
Imo the core of the problem is not here. Actually we make it more "twisted" then it was before.
By including param `gbrecs=true` we inform the api that incrementing download counts is already taken care of and api shouldn't increment them for the second time (At least that is my knowledge about this param).
So now we have situation:
 - When we download from ui we tell api that we already handled download counts (which is not true) and api will increment them regardless.
 - When we use api directly (`Datasets.java`): path `{id}/versions/{versionId}/files/download` we still not bumping download counts

Reply:
I believe the opposite situation occurs. By including `gbrecs=true` we inform api to create a guestbook response entry. 
In `FileDownloadAPIHandler.downloadFiles` we have:

```
                        if (gbrecs && file.isReleased()) {
                            GuestbookResponse gbr = guestbookResponseService.initAPIGuestbookResponse(file.getOwner(), file, session, apiTokenUser);
                            guestbookResponseService.save(gbr);
                        }
```
So we not only need `gbrecs` param but we want it to be `true` to count downloads. 
We always want to count 'whole dataset downloads` from UI.  I assumed we still want to enable api users to use `gbrecs` param - as in other endpoint.

Reply:
I believe the condition is wrong. It should be `if (!gbrecs && file.isReleased())`